salt/states/elasticsearch.py

```diff
@@ -275,10 +275,16 @@ def index_template_present(name, definition, check_definition=False):
                     ret['comment'] = 'Cannot create index template {0}, {1}'.format(name, output)
         else:
             if check_definition:
-                definition_to_diff = {'aliases': {}, 'mappings': {}, 'settings': {}}
-                definition_to_diff.update(definition)
+                if type(definition) == str:
+                    definition_parsed = salt.utils.json.loads(definition)
+                else:
+                    definition_parsed = definition
                 current_template = __salt__['elasticsearch.index_template_get'](name=name)[name]
-                diff = __utils__['dictdiffer.deep_diff'](current_template, definition_to_diff)
+                # Prune empty keys (avoid false positive diff)
+                for key in ("mappings", "aliases", "settings"):
+                    if current_template[key] == {}:
```

Comment:
Same comment here. :)

Reply:
The check here it not th know if it's a dict but and empty dict.
I find ```isinstance(current_template[key], dict) and len(current_template[key].key() == 0)``` a little bit too wordy for that purpose, any suggestion ?

Reply:
What about `if not current_template[key]:`?

Reply:
```if not current_template[key]:``` would also match non dict typed values, I'll check if it might have any border effect as if an empty string was an expected value

Reply:
You're right and this is fine. It just felt that an awkward check to me.